### PR TITLE
BN_pseudo_rand is really BN_rand

### PR DESCRIPTION
**Pull request by richsalz**
And BN_pseudo_rand_range is really BN_rand_range

I am starting to look at updating the RAND family for this release, and I started looking at bn_rand. It turns out that the "psuedo" BN rand functions don't do anything different.  So I renamed a parameter to be more clear, and replaced BN_pseudo_rand{_range} with BN_rand{_range}

### CODE REVIEW

**Comment by richsalz** [comment]
Added a commit which deprecates them.

File: include/openssl/bn.h [hunk 1]
```diff
@@ -154,9 +154,10 @@ void BN_CTX_start(BN_CTX *ctx);
 BIGNUM *BN_CTX_get(BN_CTX *ctx);
 void BN_CTX_end(BN_CTX *ctx);
 int BN_rand(BIGNUM *rnd, int bits, int top, int bottom);
-int BN_pseudo_rand(BIGNUM *rnd, int bits, int top, int bottom);
 int BN_rand_range(BIGNUM *rnd, const BIGNUM *range);
-int BN_pseudo_rand_range(BIGNUM *rnd, const BIGNUM *range);
+DEPRECATEDIN_1_1_0(int BN_pseudo_rand(BIGNUM *rnd,
+                                      int bits, int top, int bottom))
+DEPRECATEDIN_1_1_0(int BN_pseudo_rand_range(BIGNUM *rnd, const BIGNUM *range))
```
Comment (levitte) [inline include/openssl/bn.h]: Not ok, this is a change of an existing API.  You may, however, use `DEPRECATEDIN_1_2_0`, it just needs to be defined in `include/openssl/opensslconf.h.in` (I'm surprised, I thought that was already added...  by @mattcaswell if I remember correctly)

File: crypto/bn/bn_rand.c [hunk 2]
```diff
@@ -174,14 +172,9 @@ static int bn_rand_range(int pseudo, BIGNUM *r, const BIGNUM *range)
     return 1;
 }
 
-int BN_rand_range(BIGNUM *r, const BIGNUM *range)
-{
-    return bn_rand_range(0, r, range);
-}
```
Comment (levitte) [inline crypto/bn/bn_rand.c]: Errr, so where is it implemented?  It's still declared in `bn.h`!

File: crypto/bn/bn_rand.c [hunk 3]
```diff
@@ -174,14 +172,9 @@ static int bn_rand_range(int pseudo, BIGNUM *r, const BIGNUM *range)
...
-
 int BN_pseudo_rand_range(BIGNUM *r, const BIGNUM *range)
```
Comment (levitte) [inline crypto/bn/bn_rand.c]: This function needs to be guarded with:

``` C
#if (OPENSSL_API_COMPAT < 0x10200000L)
```

File: crypto/bn/bn_rand.c [hunk 4]
```diff
@@ -100,19 +100,17 @@ int BN_rand(BIGNUM *rnd, int bits, int top, int bottom)
 
 int BN_pseudo_rand(BIGNUM *rnd, int bits, int top, int bottom)
```
Comment (levitte) [inline crypto/bn/bn_rand.c]: This function needs to be guarded with:

``` C
#if (OPENSSL_API_COMPAT < 0x10200000L)
```

File: include/openssl/bn.h (hunk 1 above)
Comment (mattcaswell) [inline include/openssl/bn.h]: > (I'm surprised, I thought that was already added... by @mattcaswell if I remember correctly)

It was in #3544 which never got merged.

Comment (levitte) [inline include/openssl/bn.h]: Thanks.  So @richsalz, this is the commit to snatch: 2597a2eb44f18474b0c82338ea80e46b3b333c07

**Comment by richsalz** [comment]
updated the second commit to do the deprecation right for 1.2.0
The old static ```bn_rand_range``` was renamed to ```BN_rand_range```
Addressed all issues so far.

File: include/openssl/opensslconf.h.in [hunk 5]
```diff
@@ -95,6 +95,12 @@ extern "C" {
 # define OPENSSL_API_COMPAT OPENSSL_MIN_API
 #endif
 
+#if OPENSSL_API_COMPAT < 0x10200000L
+# define DEPRECATEDIN_1_2_0(f)   DECLARE_DEPRECATED(f)
+#else
+# define DEPRECATEDIN_1_2_0(f)
+#endif
```
Comment (kroeckx) [inline include/openssl/opensslconf.h.in]: I seem to be missing something. We already have variables marked as deprecated in 1.2.0, but I can't find how.

Comment (mattcaswell) [inline include/openssl/opensslconf.h.in]: Using OPENSSL_API_COMPAT, e.g.

````
# if OPENSSL_API_COMPAT < 0x10200000L
/*
 * LONG and ZLONG are strongly discouraged for use as stored data, as the
 * underlying C type (long) differs in size depending on the architecture.
 * They are designed with 32-bit longs in mind.
 */
DECLARE_ASN1_ITEM(LONG)
DECLARE_ASN1_ITEM(ZLONG)
# endif
````

**Comment by richsalz** [comment]
There was a PR but it’s not merged yet.  Is that what you meant?

**Comment by richsalz** [comment]
never mind prev comment

**Comment by richsalz** [comment]
ping.  we shouldn't be calling deprecated functions inside the library :)

**Comment by richsalz** [comment]
I changed the documentation to say “may be deprecated” and remove the “1.2.0 deprecation” stuff. I think the builds will pass now, let’s see.

**Comment by richsalz** [comment]
Build failures are unrelated.  I will to nudge them again.

**Comment by levitte** [comment]
A rebase on a fresher master might help

**Comment by richsalz** [comment]
Rebased and pushed.

**Comment by richsalz** [comment]
builds clean, ping @mattcaswell

**Comment by richsalz** [comment]
thanks.

**Pull request closed without merging**